### PR TITLE
Push review for Vanessa to review

### PR DESCRIPTION
**Pull request by benfluleck**
This PR has been issued against latest pull to my earliest pull on the repository
#### Description

This latest pull will show the amount of progress i have made in my Andela journey

#### What does this PR do?

This will help me become a world class developer

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New chore (expected functionality to be implemented)

#### Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: client/__test__/reducers/userReducers.spec.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import userReducer from '../../src/app/reducers/userReducers';
+
+
```
Comment (vynessa) [inline client/__test__/reducers/userReducers.spec.js:3]: Remove this line here

File: client/src/app/actions/actiontype.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 export const USER_LOGGED_IN = 'USER_LOGGED_IN';
 
+export const USER_AUTH_FAILURE = 'USER_AUTH_FAILURE';
+
 export const USER_LOGGED_OUT = 'USER_LOGGED_OUT';
 
-export const USER_SIGN_IN_FAILURE = 'USER_SIGN_IN_FAILURE';
+export const USER_SIGN_UP_FAILURE = 'USER_SIGN_UP_FAILURE';
 
 export const SIGNUP_USER_SUCCESS = 'SIGNUP_USER_SUCCESS';
```
Comment (vynessa) [inline client/src/app/actions/actiontype.js:10]: For consistency, try to make you variable names uniform. Here you have `SIGNUP_USER_SUCCESS`. With the way other variables are being named here, this should be `USER_SIGNUP_SUCCESS`

File: client/src/app/actions/api.js [hunk 3]
```diff
@@ -1,13 +1,11 @@
 import axios from 'axios';
 
-/**
- *
- */
+
```
Comment (vynessa) [inline client/src/app/actions/api.js]: Take out this line.

File: client/src/app/actions/authenticate.js [hunk 4]
```diff
@@ -1,94 +1,118 @@
-import jwtdecode from 'jwt-decode';
+import {showErrorNotification, showSuccessNotification} from './notifications';
+import {Toast} from 'react-materialize';
 
 import { USER_LOGGED_IN,
   USER_LOGGED_OUT,
-  USER_SIGN_IN_FAILURE,
   SIGNUP_USER_SUCCESS,
```
Comment (vynessa) [inline client/src/app/actions/authenticate.js:6]: Also reflect the variable name change here.

File: client/src/app/actions/authenticate.js [hunk 5]
```diff
@@ -1,94 +1,118 @@
...
 
 import api from './api';
 import setAuthorizationToken from '../utils/setAuthorizationToken';
 
-export const setCurrentUser = user =>
-  ({
-    type: SET_CURRENT_USER,
-    user
-  });
+
```
Comment (vynessa) [inline client/src/app/actions/authenticate.js:13]: Take out lines 13 and 14.

File: client/src/app/actions/authenticate.js [hunk 6]
```diff
@@ -1,94 +1,118 @@
...
-    isAuthenticated: false,
     user
   });
-export const signInUserFailure = error =>
-  ({
-    type: USER_SIGN_IN_FAILURE,
-    isAuthenticated: false,
-    error
-  });
-export const signUpUserSuccess = user => ({ type: SIGNUP_USER_SUCCESS, user });
```
Comment (vynessa) [inline client/src/app/actions/authenticate.js:38]: Take out this line as well.

File: client/src/app/actions/authenticate.js [hunk 7]
```diff
@@ -1,94 +1,118 @@
...
+ * @function userAuthFailure
+ * @param {object} response
+ * @returns {object} action: type and response
+ */
+export const userAuthFailure = (user) =>
+({
+  type: USER_AUTH_FAILURE,
+  user
+});
+
```
Comment (vynessa) [inline client/src/app/actions/authenticate.js:51]: Take out lines 51 and 52. Note that you should have one line spacing between different functions or method anything more than one should be taken out.

File: client/src/app/actions/authenticate.js [hunk 8]
```diff
@@ -1,94 +1,118 @@
...
-    if (user.status !== 200) {
-      dispatch(signInUserFailure(user));
-      Promise.reject(data);
-    } else {
-      dispatch(signUpUserSuccess(user));
-
-      return user.data;
-    }
+    dispatch(signUpUserSuccess(user));
+    dispatch(showSuccessNotification({user}));
```
Comment (vynessa) [inline client/src/app/actions/authenticate.js:78]: Great job here.

File: client/src/app/actions/uploadImage.js [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+.then((response)=>{
+  dispatch(UploadImageToCloud(response.body))
+  return (response.body)
+}).catch(error=>{
+  UploadImageToCloudFailure(error)
+});
+}
+
+
+//  export const imageUploadToDb = ()=> (dispatch)=>
```
Comment (vynessa) [inline client/src/app/actions/uploadImage.js]: Zombie code. Kill it.

File: client/src/app/components/container/authentication/SignUpPage.jsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * handles registartion of users
+ * @class SignUpPage
+ * @extends {React.Component}
+ */
+class SignUpPage extends React.Component {
+  submit = (data)=> {
+  this.props.
+    signup(data)
+.then((res)=>{
```
Comment (vynessa) [inline client/src/app/components/container/authentication/SignUpPage.jsx:17]: Indent this file well. Also take out extra lines where not needed

File: client/src/app/components/container/dashboard/presentation/Dashboard.jsx [hunk 11]
```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { Row } from 'react-materialize';
+import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
+import SideNav from '../../../presentation/common/SideNav/index.jsx';
+import DisplayAllBorrowedBooks from '../../booklist/DisplayAllBorrowedBooks.jsx';
+import DisplayAllBooks from '../../booklist/DisplayAllBooks.jsx';
+// import UploadModal from '../../../presentation/common/modal/UploadModal.jsx';
```
Comment (vynessa) [inline client/src/app/components/container/dashboard/presentation/Dashboard.jsx]: Zombie code. Try not to push this commented out code. Instead, if you feel you would need it later, keep it in the note app and reference the file name there.

File: client/src/app/components/presentation/authentication/SignUpPage.jsx [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+   */
+  onSubmit = (e) => {
+    e.preventDefault();
+    if (this.isValid()) {
+      this.setState({errors: {}})
+      this
+        .props
+        .submit(this.state)
+    }
+  }
```
Comment (vynessa) [inline client/src/app/components/presentation/authentication/SignUpPage.jsx:84]: Add a new line after this.

File: client/src/app/components/presentation/common/Preloader/ShowProgressBar.jsx [hunk 13]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import {Col, Row, ProgressBar} from 'react-materialize';
+
+const ShowProgressBar = () => (
+  <Row>
+
```
Comment (vynessa) [inline client/src/app/components/presentation/common/Preloader/ShowProgressBar.jsx]: Take out this line

File: client/src/app/components/presentation/common/Preloader/ShowProgressBar.jsx [hunk 14]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import {Col, Row, ProgressBar} from 'react-materialize';
+
+const ShowProgressBar = () => (
+  <Row>
+    
+      <Col l={4} s={4} m={4}>
+          Loading....
+        <ProgressBar/>
+
```
Comment (vynessa) [inline client/src/app/components/presentation/common/Preloader/ShowProgressBar.jsx]: Do the same for this.

File: client/src/app/components/presentation/common/SideNav/index.jsx [hunk 15]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import UserView from './UserView.jsx';
+
```
Comment (vynessa) [inline client/src/app/components/presentation/common/SideNav/index.jsx:4]: Take out this line.

File: client/src/app/components/presentation/common/book/DisplayBook.jsx [hunk 16]
```diff
@@ -1,45 +1,56 @@
...
-
-
-
-}
+const Book = books => (
+
+  <div className="col l3">
+    <ReactTooltip />
+    <div className="card" >
+      {/* <a
```
Comment (vynessa) [inline client/src/app/components/presentation/common/book/DisplayBook.jsx]: Zombie code. You know what to do.

File: client/src/app/components/presentation/common/modal/UploadModal.jsx [hunk 17]
```diff
@@ -0,0 +1,196 @@
...
+
+
+
+  onClick(event){
+    event.preventDefault();
+    this.setState({isLoading: true});
+    if(this.state.filename)
+      {
+        console.log('I am here', this.props.secure_url)
+        // imageUploadToDb
```
Comment (vynessa) [inline client/src/app/components/presentation/common/modal/UploadModal.jsx:81]: Zombie code. I will like you to replicate killing this wherever you have such existing.

File: client/src/app/components/presentation/messages/NoBooksMessages.jsx [hunk 18]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+
+/**
+ * @description Component for Message for No Books
+ * @class NoBooksMessage
+ */
+const NoBooksMessage = () => (
```
Comment (vynessa) [inline client/src/app/components/presentation/messages/NoBooksMessages.jsx:8]: I think this function could do with a better name.

**Review by vynessa — COMMENTED** [review]
Good job so far. Kindly address my comments and reply where necessary.

File: client/src/app/css/style.scss [hunk 19]
```diff
@@ -96,17 +153,156 @@ footer {
...
+.modal-content{
+	padding-top:7px;
+	padding-right:0 !important;
+	padding-left:0 !important;
+	padding-top:0 !important;
+}
+
+.react-tabs__tab-list {
+	padding-top: 15px;
+	font-family: $font-stack;
```
[the remark this answers is not in the document]
Comment (benfluleck) [reply client/src/app/css/style.scss:314]: Installed .sass-lint.yml to standardize my sass files

File: client/src/app/css/style.scss [hunk 20]
```diff
@@ -96,17 +153,156 @@ footer {
...
+}
+
+.react-tabs__tab--selected {
+	background: $base-orange;
+	color: $base-white;
+}
+
+.main-text {
+	background-color: white;
```
[the remark this answers is not in the document]
Comment (benfluleck) [reply client/src/app/css/style.scss:323]: changed to the base white variable

**Pull request closed without merging**